lib/util/state.js

```diff
@@ -141,24 +118,69 @@ export function getActiveError (state) {
  * Get the appropriate message for a trip planning error given possibly pre-
  * configured message overrides from the config.
  */
-export function getErrorMessage (error, errorMessages) {
+export function getErrorMessage (error, errorMessages, locale) {
   if (!error) return null
 
-  // use the message provided by OTP by default
-  let message = error.msg
+  let message
+  if (error.network) {
+    message = (
+      <FormattedMessage
+        id='util.state.networkUnavailable'
+        values={{ network: error.network }}
+      />
+    )
+  } else if (error.id === 404) {
+    // Localized messages for OTP's 404 PATH_NOT_FOUND error.
+    // include the modes if doing batch routing.
+    const noTripFoundText = (error.showModes && error.modes.length > 0)
+      ? (
+        <FormattedMessage
+          id='util.state.noTripFoundForMode'
+          values={{
+            modes: (
+              <FormattedList
+                type='conjunction'
+                value={error.modes.map(
+                  // Styled component makes text lowercase
+                  mode => <StyledFormattedMode key={mode} mode={mode.toLowerCase()} />
+                )}
+              />
+            )
+          }}
+        />
+      )
+      : <FormattedMessage id='util.state.noTripFound' />
+
+    message = (
+      <FormattedMessage
+        id='util.state.noTripFoundWithReason'
+        values={{
+          noTripFound: noTripFoundText,
+          reason: <FormattedMessage id='util.state.noTripFoundReason' />
+        }}
+      />
+    )
+  }
+
+  message = message || <FormattedMessage id='util.state.errorPlanningTrip' />
 
   // check for configuration-defined message override
   if (errorMessages) {
     const msgConfig = errorMessages.find(m => m.id === error.id)
     if (msgConfig) {
-      if (error.modes && msgConfig.modes) {
-        for (const mode of msgConfig.modes) {
-          if (error.modes.includes(mode)) {
-            message = msgConfig.msg
+      const lang = locale?.split('-')[0]
+      const { modes, msg } = msgConfig
```

Comment:
I think if we switched to the other way of putting the error message overrides in the config this could be done using g the normal intl methods! 

See https://github.com/opentripplanner/otp-react-redux/pull/456/files#diff-9fd873d61d8a70d6eaf49fe78959e408c8bf2314eb4e645e048cfd98b8a43916R281

Reply:
Will update this in a PR just addressing the localized error messages